src/Sema.zig

```diff
@@ -17974,6 +17975,101 @@ fn zirWasmMemoryGrow(
+
+    if (try sema.lookupInNamespace(block, name_src, namespace, decl_name, true)) |decl_index| {
+        try sema.mod.ensureDeclAnalyzed(decl_index);
+
+        const decl = sema.mod.declPtr(decl_index);
+        const decl_tv = try decl.typedValue();
+
+        // Make sure declaration is in scope and is marked public
+        if (!decl.is_pub and decl.getFileScope() != block.getFileScope()) {
+            const msg = msg: {
+                const msg = try sema.errMsg(block, src, "'{s}' is not marked 'pub'", .{
+                    decl_name,
+                });
+                errdefer msg.destroy(sema.gpa);
+
+                try sema.mod.errNoteNonLazy(decl.srcLoc(), msg, "declared here", .{});
+                break :msg msg;
+            };
+            return sema.failWithOwnedErrorMsg(block, msg);
+        }
+
+        const opt_linksection = if (decl.@"linksection") |ls| blk: {
+            var anon_decl = try block.startAnonDecl(src);
+            defer anon_decl.deinit();
+            const @"linksection" = std.mem.span(ls);
+            const bytes = try anon_decl.arena().dupe(u8, @"linksection"[0 .. @"linksection".len + 1]);
+            const new_decl = try anon_decl.finish(
+                try Type.Tag.array_u8_sentinel_0.create(anon_decl.arena(), bytes.len - 1),
+                try Value.Tag.bytes.create(anon_decl.arena(), bytes),
+                0, // default alignment
+            );
+            break :blk try Value.Tag.decl_ref.create(sema.arena, new_decl);
+        } else null;
+
+        const field_values = try sema.arena.alloc(Value, 3);
+
+        // ty: type
+        field_values[0] = try Value.Tag.ty.create(sema.arena, try decl_tv.ty.copy(sema.arena));
+
+        // alignment: comptime_int
+        field_values[1] = if (decl.@"align" != 0)
+            try Value.Tag.int_u64.create(sema.arena, decl.@"align")
+        else
+            try decl.ty.lazyAbiAlignment(target, sema.arena);
+
+        // link_section: ?[:0]const u8
+        field_values[2] = try sema.optRefValue(
+            block,
+            src,
+            Type.initTag(.const_slice_u8_sentinel_0),
+            opt_linksection,
+        );
+
+        return sema.addConstant(
+            try sema.getBuiltinType(block, src, "DeclarationInfo"),
+            try Value.Tag.aggregate.create(sema.arena, field_values),
+        );
+    }
+
+    // Failed to lookup declaration in namespace
+    return sema.failWithOwnedErrorMsg(block, msg: {
+        const namespace_name = try namespace_ty.nameAllocArena(sema.arena, sema.mod);
+
+        const msg = try sema.errMsg(
+            block,
+            name_src,
+            "declaration of '{s}' could not be found in type '{s}'",
+            .{ decl_name, namespace_name },
+        );
+        errdefer msg.destroy(sema.gpa);
+
+        try sema.mod.errNoteNonLazy(namespace_ty.declSrcLoc(sema.mod), msg, "declared here", .{});
+        break :msg msg;
+    });
+}
```

Comment:
Use `failWithBadMemberAccess` preferably with `orelse` to avoid indenting the rest of the function.